share/github-backup-utils/ghe-rsync

```diff
@@ -13,15 +13,30 @@ set -o pipefail
 # Filter vanished file warnings from both stdout (rsync versions < 3.x) and
 # stderr (rsync versions >= 3.x). The complex redirections are necessary to
 # filter stderr while also keeping stdout and stderr separated.
-IGNOREOUT='^(file has vanished: |rsync warning: some files vanished before they could be transferred)'
-(rsync "${@}" $GHE_EXTRA_RSYNC_OPTS 3>&1 1>&2 2>&3 3>&- |
-  (egrep -v "$IGNOREOUT" || true)) 3>&1 1>&2 2>&3 3>&- |
-  (egrep -v "$IGNOREOUT" || true)
+ignoreout='^(file has vanished: |rsync warning: some files vanished before they could be transferred)'
+
+# Remove --ignore-missing-args parameter if using an earlier version of rsync.
+if ! rsync -h | grep -q '\-\-ignore-missing-args'; then
+  for parameter; do
```

Comment:
Can you add the `in "$@"` part here too to make it clear this is what we're iterating over - it's not obviously clear unless you know about this behaviour.